pkg/serverless/serverless.go

```diff
@@ -129,6 +129,8 @@ func WaitForNextInvocation(stopCh chan struct{}, daemon *daemon.Daemon, id regis
 		functionArn := removeQualifierFromArn(payload.InvokedFunctionArn)
 		callInvocationHandler(daemon, functionArn, payload.DeadlineMs, safetyBufferTimeout, payload.RequestID, handleInvocation)
 	} else if payload.EventType == Shutdown {
+		// Ensure we start log collection during a shutdown event too in case an invoke event is never received
+		daemon.StartLogCollection()
```

Comment:
we can do that thanks to the `process_once` right?
if yes, maybe a comment on that (or better a test) would be helpful to make sure we don't end up with a code which could potentially start the log collection twice?

Reply:
yup, that's right. Good point ✔️

Reply:
I edited the comment and also made a test to check that the execution context isn't reset when multiple calls to StartLogCollection() are made. Sort of a unique test case: https://github.com/DataDog/datadog-agent/pull/18748/files#diff-8f587b647d9d9ee523a2f5d4613317e2a1ad711f147b2c8ac153003beaac2512R1291